src/pages/CoinPage.jsx

```diff
@@ -1,8 +1,138 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import { makeStyles, Typography, LinearProgress } from '@material-ui/core';
+import axios from 'axios';
+import ReactHtmlParser from 'react-html-parser';
+import { SingleCoin } from '../config/api';
+import { CryptoState } from '../CryptoContext';
+import { numberWithCommas } from './Banner/Carousel';
+import CoinsInfo from '../components/CoinsInfo';
+
+const useStyles = makeStyles((theme) => ({
+  container: {
+    display: 'flex',
+    [theme.breakpoints.down('md')]: {
+      flexDirection: 'column',
+      alignItems: 'center',
+    },
+  },
+  sidebar: {
+    width: '30%',
+    [theme.breakpoints.down('md')]: {
+      width: '100%',
+    },
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+    marginTop: 25,
+    borderRight: '2px solid grey',
+  },
+  heading: {
+    fontWeight: 'bold',
+    marginBottom: 20,
+    fontFamily: 'Montserrat',
+  },
+  description: {
+    width: '100%',
+    fontFamily: 'Montserrat',
+    padding: 25,
+    paddingBottom: 15,
+    paddingTop: 0,
+    textAlign: 'justify',
+  },
+
+}));
 
 const CoinPage = () => {
+  const { id } = useParams();
+  const [coin, setCoin] = useState();
```

Comment:
I'd recommend destructuring `coin` to streamline your JSX code:
```suggestion
  const [coin, setCoin] = useState();
  const { image, name, description, market_cap_rank, market_data } = coin;
```